src/QsCompiler/CompilerOptimizations/ConstantPropagation.fs

```diff
@@ -71,8 +60,68 @@ type ConstantPropagator(compiledCallables: ImmutableDictionary<QsQualifiedName,
                 if stm.Kind = ImmutableBinding then
                     if isLiteral rhs.Expression callableDict then
                         fillVars vars (StringTuple.fromSymbolTuple lhs, rhs.Expression)
-                        declarations <- declarations @ [sprintf "%O = %O" (StringTuple.fromSymbolTuple lhs) (prettyPrint rhs.Expression)]
                         // printfn "Found constant declaration: %O = %O" (StringTuple.fromSymbolTuple lhs) (prettyPrint rhs.Expression)
                 QsBinding<TypedExpression>.New stm.Kind (lhs, rhs) |> QsVariableDeclaration
+
+            override statementKind.onConditionalStatement stm = 
+                let cbList, cbListEnd =
+                    stm.ConditionalBlocks |> Seq.fold (fun s (c, b) ->
+                        let cond, block = statementKind.onPositionedBlock (Some c, b)
+                        match cond.Value.Expression with
+                        | BoolLiteral true -> s @ [None, block]
+                        | BoolLiteral false -> s
+                        | _ -> s @ [cond, block]
+                    ) [] |> List.ofSeq |> takeWhilePlus1 (fun (c, b) -> c <> None)
+                let defaultCase = stm.Default |> QsNullable<_>.Map (fun b -> statementKind.onPositionedBlock (None, b) |> snd)
+
+                let newDefault =
+                    match cbListEnd, defaultCase with
+                    | Some (_, a), _ -> Value a
+                    | _, Value a -> Value a
+                    | _ -> Null
+                match cbList, newDefault with
+                | [], Value x ->
+                    x.Body |> QsScopeStatement.New |> QsScopeStatement
+                | [], Null ->
+                    QsScope.New ([], LocalDeclarations.New []) |> QsScopeStatement.New |> QsScopeStatement
```

Comment:
You need to properly eliminate the statement in that case entirely.

Reply:
Issue is fixed in later commit